src/qt/askpassphrasedialog.cpp

```diff
@@ -216,18 +218,57 @@ void AskPassphraseDialog::textChanged()
     {
     case Encrypt: // New passphrase x2
         acceptable = !ui->passEdit2->text().isEmpty() && !ui->passEdit3->text().isEmpty();
+        this->updatePasswordStrengthIndicator(ui->passEdit2);
         break;
     case Unlock: // Old passphrase x1
     case Decrypt:
         acceptable = !ui->passEdit1->text().isEmpty();
         break;
     case ChangePass: // Old passphrase x1, new passphrase x2
         acceptable = !ui->passEdit1->text().isEmpty() && !ui->passEdit2->text().isEmpty() && !ui->passEdit3->text().isEmpty();
+        this->updatePasswordStrengthIndicator(ui->passEdit2);
         break;
     }
     ui->buttonBox->button(QDialogButtonBox::Ok)->setEnabled(acceptable);
 }
 
+void AskPassphraseDialog::updatePasswordStrengthIndicator(QLineEdit *lineedit)
+{
+    std::string pass = lineedit->text().toUtf8().constData();
+    bool mixedcase = (boost::to_upper_copy<std::string>(pass) != pass) && (boost::to_lower_copy<std::string>(pass) != pass);
+    bool specialChars = pass.find_first_not_of("ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxzy123456789") != std::string::npos;
+    bool digits = pass.find_first_not_of("1234567890") != std::string::npos;
+    // Do nothing if the PW was empty
+    if (pass.size() == 0) {
+        ui->labelSecurityLevel->setText("");
+        return;
+    }
+    // Passwords smaller than 10 chars are generally very weak
+    if (pass.size() < 10) {
+        ui->labelSecurityLevel->setText("Very weak");
+        ui->labelSecurityLevel->setStyleSheet("QLabel {color: #ff0000; }");
+        return;
+    }
+    // Passwords with no mixed case or special characters but >8 are weak
```

Comment:
nit 10, not 8

Reply:
hebasto already mentioned it. Will update it